From d350f4b7deb03cbba346acfae46bc51ebc9d7444 Mon Sep 17 00:00:00 2001
From: Jan Gutsche <34797331+jaagut@users.noreply.github.com>
Date: Wed, 11 Oct 2023 21:38:20 +0200
Subject: [PATCH] Fix typeguard error (#149)

For typeguard==3.0.2, `default_value: any` raised a `TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union` during colcon build of a Python node using this library.
Using the `Any` from the `typing` module fixed this issue.

Yes, this problem does not occur with the typeguard version provided by apt (2.2.2), but could also occur if Ubuntu bumps that version or for other users with a newer local pip install of typeguard.
---
 .../parse_yaml.py               | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/generate_parameter_library_py/parse_yaml.py b/generate_parameter_library_py/parse_yaml.py
index cf317ec..7ae898a 100644
--- a/generate_parameter_library_py/parse_yaml.py
+++ b/generate_parameter_library_py/parse_yaml.py
@@ -31,7 +31,7 @@
 
 from jinja2 import Template
 from typeguard import typechecked
-from typing import List, Optional
+from typing import Any, List, Optional
 from yaml.parser import ParserError
 from yaml.scanner import ScannerError
 import os
@@ -147,7 +147,7 @@ class CodeGenVariableBase:
         name: str,
         param_name: str,
         defined_type: str,
-        default_value: any,
+        default_value: Any,
     ):
         if language == "cpp":
             self.conversation = CPPConverstions()
-- 
2.41.0

